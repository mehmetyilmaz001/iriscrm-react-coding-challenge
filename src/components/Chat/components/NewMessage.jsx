
import PropTypes from 'prop-types';
import { InputBase, IconButton, Paper } from "@mui/material";
import SendIcon from '@mui/icons-material/Send';

function NewMessage(props) {
    
  const _onMessageEnter = (message) => {
    props.onEnterMessage(message);
  }  

  return (
    <Paper
      className="chat-footer"
      component="form"
      sx={{ p: "2px 4px", display: "flex", alignItems: "center", width: '100%' }}
      onSubmit={(e) => {
          e.preventDefault();
          const val =  e.target.message.value;
          if(val) {
              _onMessageEnter(val);
              e.target.message.value = '';
          }
        }
    }
    >
      <InputBase
        sx={{ ml: 1, flex: 1 }}
        id="message"
        placeholder="Enter message text here..."
        autoComplete='off'
        inputProps={{ "aria-label": "Enter message text here" }}
      />
      <IconButton type="submit" sx={{ p: "10px" }} aria-label="Send">
        <SendIcon />
      </IconButton>
    </Paper>
  );
}

export default NewMessage;


NewMessage.prototype = {
    onEnterMessage: PropTypes.func,
}
