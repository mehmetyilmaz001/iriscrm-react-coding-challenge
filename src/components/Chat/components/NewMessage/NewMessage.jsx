
import PropTypes from 'prop-types';
import { InputBase, IconButton, Paper } from "@mui/material";
import SendIcon from '@mui/icons-material/Send';

function NewMessage(props) {
    
  const _onMessageEnter = (message) => {
    props.onEnterMessage(message);
  }  

  return (
    <Paper
      data-testid="new-message"
      className="chat-footer"
      component="form"
      id="new-message"
      sx={{ p: "2px 4px", display: "flex", alignItems: "center", width: '100%' }}
      onSubmit={(e) => {
          e.preventDefault();
          console.log(e.target);
          const val =  e.target.message.value;
          if(val) {
              _onMessageEnter(val);
              e.target.message.value = '';
          }
        }
    }
    >
      <InputBase
        sx={{ ml: 1, flex: 1 }}
        id="message"
        
        placeholder="Enter message text here..."
        autoComplete='off'
        inputProps={{ "aria-label": "Enter message text here", "data-testid":"message" }}
      />
      <IconButton type="submit" data-testid="send-message" sx={{ p: "10px" }} aria-label="Send">
        <SendIcon />
      </IconButton>
    </Paper>
  );
}

export default NewMessage;


NewMessage.prototype = {
    onEnterMessage: PropTypes.func,
}
