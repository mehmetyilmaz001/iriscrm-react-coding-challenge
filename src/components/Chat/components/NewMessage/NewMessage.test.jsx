import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import NewMessage from './NewMessage';

describe('NewMessage component', () => {
    const onEnterMessage = jest.fn();

    const component = mount(<NewMessage onEnterMessage={onEnterMessage}/>);

    it('expect to render component', () => {
        expect(toJson(component)).toMatchSnapshot();
    });
});