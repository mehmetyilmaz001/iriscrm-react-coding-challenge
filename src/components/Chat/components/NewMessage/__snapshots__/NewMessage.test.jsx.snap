// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewMessage component expect to render component 1`] = `
<NewMessage
  onEnterMessage={[MockFunction]}
>
  <ForwardRef(Paper)
    className="chat-footer"
    component="form"
    data-testid="new-message"
    id="new-message"
    onSubmit={[Function]}
    sx={
      Object {
        "alignItems": "center",
        "display": "flex",
        "p": "2px 4px",
        "width": "100%",
      }
    }
  >
    <MuiPaperRoot
      as="form"
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 chat-footer"
      data-testid="new-message"
      id="new-message"
      onSubmit={[Function]}
      ownerState={
        Object {
          "children": Array [
            <ForwardRef(InputBase)
              autoComplete="off"
              id="message"
              inputProps={
                Object {
                  "aria-label": "Enter message text here",
                  "data-testid": "message",
                }
              }
              placeholder="Enter message text here..."
              sx={
                Object {
                  "flex": 1,
                  "ml": 1,
                }
              }
            />,
            <ForwardRef(IconButton)
              aria-label="Send"
              data-testid="send-message"
              sx={
                Object {
                  "p": "10px",
                }
              }
              type="submit"
            >
              <Memo />
            </ForwardRef(IconButton)>,
          ],
          "className": "chat-footer",
          "component": "form",
          "data-testid": "new-message",
          "elevation": 1,
          "id": "new-message",
          "onSubmit": [Function],
          "square": false,
          "sx": Object {
            "alignItems": "center",
            "display": "flex",
            "p": "2px 4px",
            "width": "100%",
          },
          "variant": "elevation",
        }
      }
      sx={
        Object {
          "alignItems": "center",
          "display": "flex",
          "p": "2px 4px",
          "width": "100%",
        }
      }
    >
      <Noop />
      <form
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 chat-footer css-1gqfvv-MuiPaper-root"
        data-testid="new-message"
        id="new-message"
        onSubmit={[Function]}
      >
        <ForwardRef(InputBase)
          autoComplete="off"
          id="message"
          inputProps={
            Object {
              "aria-label": "Enter message text here",
              "data-testid": "message",
            }
          }
          placeholder="Enter message text here..."
          sx={
            Object {
              "flex": 1,
              "ml": 1,
            }
          }
        >
          <GlobalStyles
            styles={
              Object {
                "@keyframes mui-auto-fill": Object {
                  "from": Object {
                    "display": "block",
                  },
                },
                "@keyframes mui-auto-fill-cancel": Object {
                  "from": Object {
                    "display": "block",
                  },
                },
              }
            }
          >
            <GlobalStyles
              defaultTheme={
                Object {
                  "breakpoints": Object {
                    "between": [Function],
                    "down": [Function],
                    "keys": Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ],
                    "not": [Function],
                    "only": [Function],
                    "unit": "px",
                    "up": [Function],
                    "values": Object {
                      "lg": 1200,
                      "md": 900,
                      "sm": 600,
                      "xl": 1536,
                      "xs": 0,
                    },
                  },
                  "components": Object {},
                  "direction": "ltr",
                  "mixins": Object {
                    "toolbar": Object {
                      "@media (min-width:0px) and (orientation: landscape)": Object {
                        "minHeight": 48,
                      },
                      "@media (min-width:600px)": Object {
                        "minHeight": 64,
                      },
                      "minHeight": 56,
                    },
                  },
                  "palette": Object {
                    "action": Object {
                      "activatedOpacity": 0.12,
                      "active": "rgba(0, 0, 0, 0.54)",
                      "disabled": "rgba(0, 0, 0, 0.26)",
                      "disabledBackground": "rgba(0, 0, 0, 0.12)",
                      "disabledOpacity": 0.38,
                      "focus": "rgba(0, 0, 0, 0.12)",
                      "focusOpacity": 0.12,
                      "hover": "rgba(0, 0, 0, 0.04)",
                      "hoverOpacity": 0.04,
                      "selected": "rgba(0, 0, 0, 0.08)",
                      "selectedOpacity": 0.08,
                    },
                    "augmentColor": [Function],
                    "background": Object {
                      "default": "#fff",
                      "paper": "#fff",
                    },
                    "common": Object {
                      "black": "#000",
                      "white": "#fff",
                    },
                    "contrastThreshold": 3,
                    "divider": "rgba(0, 0, 0, 0.12)",
                    "error": Object {
                      "contrastText": "#fff",
                      "dark": "#c62828",
                      "light": "#ef5350",
                      "main": "#d32f2f",
                    },
                    "getContrastText": [Function],
                    "grey": Object {
                      "100": "#f5f5f5",
                      "200": "#eeeeee",
                      "300": "#e0e0e0",
                      "400": "#bdbdbd",
                      "50": "#fafafa",
                      "500": "#9e9e9e",
                      "600": "#757575",
                      "700": "#616161",
                      "800": "#424242",
                      "900": "#212121",
                      "A100": "#f5f5f5",
                      "A200": "#eeeeee",
                      "A400": "#bdbdbd",
                      "A700": "#616161",
                    },
                    "info": Object {
                      "contrastText": "#fff",
                      "dark": "#01579b",
                      "light": "#03a9f4",
                      "main": "#0288d1",
                    },
                    "mode": "light",
                    "primary": Object {
                      "contrastText": "#fff",
                      "dark": "#1565c0",
                      "light": "#42a5f5",
                      "main": "#1976d2",
                    },
                    "secondary": Object {
                      "contrastText": "#fff",
                      "dark": "#7b1fa2",
                      "light": "#ba68c8",
                      "main": "#9c27b0",
                    },
                    "success": Object {
                      "contrastText": "#fff",
                      "dark": "#1b5e20",
                      "light": "#4caf50",
                      "main": "#2e7d32",
                    },
                    "text": Object {
                      "disabled": "rgba(0, 0, 0, 0.38)",
                      "primary": "rgba(0, 0, 0, 0.87)",
                      "secondary": "rgba(0, 0, 0, 0.6)",
                    },
                    "tonalOffset": 0.2,
                    "warning": Object {
                      "contrastText": "#fff",
                      "dark": "#e65100",
                      "light": "#ff9800",
                      "main": "#ed6c02",
                    },
                  },
                  "shadows": Array [
                    "none",
                    "0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12)",
                    "0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12)",
                    "0px 3px 3px -2px rgba(0,0,0,0.2),0px 3px 4px 0px rgba(0,0,0,0.14),0px 1px 8px 0px rgba(0,0,0,0.12)",
                    "0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12)",
                    "0px 3px 5px -1px rgba(0,0,0,0.2),0px 5px 8px 0px rgba(0,0,0,0.14),0px 1px 14px 0px rgba(0,0,0,0.12)",
                    "0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12)",
                    "0px 4px 5px -2px rgba(0,0,0,0.2),0px 7px 10px 1px rgba(0,0,0,0.14),0px 2px 16px 1px rgba(0,0,0,0.12)",
                    "0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12)",
                    "0px 5px 6px -3px rgba(0,0,0,0.2),0px 9px 12px 1px rgba(0,0,0,0.14),0px 3px 16px 2px rgba(0,0,0,0.12)",
                    "0px 6px 6px -3px rgba(0,0,0,0.2),0px 10px 14px 1px rgba(0,0,0,0.14),0px 4px 18px 3px rgba(0,0,0,0.12)",
                    "0px 6px 7px -4px rgba(0,0,0,0.2),0px 11px 15px 1px rgba(0,0,0,0.14),0px 4px 20px 3px rgba(0,0,0,0.12)",
                    "0px 7px 8px -4px rgba(0,0,0,0.2),0px 12px 17px 2px rgba(0,0,0,0.14),0px 5px 22px 4px rgba(0,0,0,0.12)",
                    "0px 7px 8px -4px rgba(0,0,0,0.2),0px 13px 19px 2px rgba(0,0,0,0.14),0px 5px 24px 4px rgba(0,0,0,0.12)",
                    "0px 7px 9px -4px rgba(0,0,0,0.2),0px 14px 21px 2px rgba(0,0,0,0.14),0px 5px 26px 4px rgba(0,0,0,0.12)",
                    "0px 8px 9px -5px rgba(0,0,0,0.2),0px 15px 22px 2px rgba(0,0,0,0.14),0px 6px 28px 5px rgba(0,0,0,0.12)",
                    "0px 8px 10px -5px rgba(0,0,0,0.2),0px 16px 24px 2px rgba(0,0,0,0.14),0px 6px 30px 5px rgba(0,0,0,0.12)",
                    "0px 8px 11px -5px rgba(0,0,0,0.2),0px 17px 26px 2px rgba(0,0,0,0.14),0px 6px 32px 5px rgba(0,0,0,0.12)",
                    "0px 9px 11px -5px rgba(0,0,0,0.2),0px 18px 28px 2px rgba(0,0,0,0.14),0px 7px 34px 6px rgba(0,0,0,0.12)",
                    "0px 9px 12px -6px rgba(0,0,0,0.2),0px 19px 29px 2px rgba(0,0,0,0.14),0px 7px 36px 6px rgba(0,0,0,0.12)",
                    "0px 10px 13px -6px rgba(0,0,0,0.2),0px 20px 31px 3px rgba(0,0,0,0.14),0px 8px 38px 7px rgba(0,0,0,0.12)",
                    "0px 10px 13px -6px rgba(0,0,0,0.2),0px 21px 33px 3px rgba(0,0,0,0.14),0px 8px 40px 7px rgba(0,0,0,0.12)",
                    "0px 10px 14px -6px rgba(0,0,0,0.2),0px 22px 35px 3px rgba(0,0,0,0.14),0px 8px 42px 7px rgba(0,0,0,0.12)",
                    "0px 11px 14px -7px rgba(0,0,0,0.2),0px 23px 36px 3px rgba(0,0,0,0.14),0px 9px 44px 8px rgba(0,0,0,0.12)",
                    "0px 11px 15px -7px rgba(0,0,0,0.2),0px 24px 38px 3px rgba(0,0,0,0.14),0px 9px 46px 8px rgba(0,0,0,0.12)",
                  ],
                  "shape": Object {
                    "borderRadius": 4,
                  },
                  "spacing": [Function],
                  "transitions": Object {
                    "create": [Function],
                    "duration": Object {
                      "complex": 375,
                      "enteringScreen": 225,
                      "leavingScreen": 195,
                      "short": 250,
                      "shorter": 200,
                      "shortest": 150,
                      "standard": 300,
                    },
                    "easing": Object {
                      "easeIn": "cubic-bezier(0.4, 0, 1, 1)",
                      "easeInOut": "cubic-bezier(0.4, 0, 0.2, 1)",
                      "easeOut": "cubic-bezier(0.0, 0, 0.2, 1)",
                      "sharp": "cubic-bezier(0.4, 0, 0.6, 1)",
                    },
                    "getAutoHeightDuration": [Function],
                  },
                  "typography": Object {
                    "body1": Object {
                      "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                      "fontSize": "1rem",
                      "fontWeight": 400,
                      "letterSpacing": "0.00938em",
                      "lineHeight": 1.5,
                    },
                    "body2": Object {
                      "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                      "fontSize": "0.875rem",
                      "fontWeight": 400,
                      "letterSpacing": "0.01071em",
                      "lineHeight": 1.43,
                    },
                    "button": Object {
                      "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                      "fontSize": "0.875rem",
                      "fontWeight": 500,
                      "letterSpacing": "0.02857em",
                      "lineHeight": 1.75,
                      "textTransform": "uppercase",
                    },
                    "caption": Object {
                      "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                      "fontSize": "0.75rem",
                      "fontWeight": 400,
                      "letterSpacing": "0.03333em",
                      "lineHeight": 1.66,
                    },
                    "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                    "fontSize": 14,
                    "fontWeightBold": 700,
                    "fontWeightLight": 300,
                    "fontWeightMedium": 500,
                    "fontWeightRegular": 400,
                    "h1": Object {
                      "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                      "fontSize": "6rem",
                      "fontWeight": 300,
                      "letterSpacing": "-0.01562em",
                      "lineHeight": 1.167,
                    },
                    "h2": Object {
                      "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                      "fontSize": "3.75rem",
                      "fontWeight": 300,
                      "letterSpacing": "-0.00833em",
                      "lineHeight": 1.2,
                    },
                    "h3": Object {
                      "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                      "fontSize": "3rem",
                      "fontWeight": 400,
                      "letterSpacing": "0em",
                      "lineHeight": 1.167,
                    },
                    "h4": Object {
                      "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                      "fontSize": "2.125rem",
                      "fontWeight": 400,
                      "letterSpacing": "0.00735em",
                      "lineHeight": 1.235,
                    },
                    "h5": Object {
                      "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                      "fontSize": "1.5rem",
                      "fontWeight": 400,
                      "letterSpacing": "0em",
                      "lineHeight": 1.334,
                    },
                    "h6": Object {
                      "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                      "fontSize": "1.25rem",
                      "fontWeight": 500,
                      "letterSpacing": "0.0075em",
                      "lineHeight": 1.6,
                    },
                    "htmlFontSize": 16,
                    "overline": Object {
                      "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                      "fontSize": "0.75rem",
                      "fontWeight": 400,
                      "letterSpacing": "0.08333em",
                      "lineHeight": 2.66,
                      "textTransform": "uppercase",
                    },
                    "pxToRem": [Function],
                    "subtitle1": Object {
                      "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                      "fontSize": "1rem",
                      "fontWeight": 400,
                      "letterSpacing": "0.00938em",
                      "lineHeight": 1.75,
                    },
                    "subtitle2": Object {
                      "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                      "fontSize": "0.875rem",
                      "fontWeight": 500,
                      "letterSpacing": "0.00714em",
                      "lineHeight": 1.57,
                    },
                  },
                  "zIndex": Object {
                    "appBar": 1100,
                    "drawer": 1200,
                    "mobileStepper": 1000,
                    "modal": 1300,
                    "snackbar": 1400,
                    "speedDial": 1050,
                    "tooltip": 1500,
                  },
                }
              }
              styles={
                Object {
                  "@keyframes mui-auto-fill": Object {
                    "from": Object {
                      "display": "block",
                    },
                  },
                  "@keyframes mui-auto-fill-cancel": Object {
                    "from": Object {
                      "display": "block",
                    },
                  },
                }
              }
            >
              <EmotionGlobal
                styles={
                  Object {
                    "@keyframes mui-auto-fill": Object {
                      "from": Object {
                        "display": "block",
                      },
                    },
                    "@keyframes mui-auto-fill-cancel": Object {
                      "from": Object {
                        "display": "block",
                      },
                    },
                  }
                }
              />
            </GlobalStyles>
          </GlobalStyles>
          <MuiInputBaseRoot
            className="MuiInputBase-root MuiInputBase-colorPrimary"
            onClick={[Function]}
            ownerState={
              Object {
                "autoComplete": "off",
                "color": "primary",
                "disabled": undefined,
                "endAdornment": undefined,
                "error": undefined,
                "focused": false,
                "formControl": undefined,
                "fullWidth": false,
                "hiddenLabel": undefined,
                "id": "message",
                "inputProps": Object {
                  "aria-label": "Enter message text here",
                  "data-testid": "message",
                },
                "multiline": false,
                "placeholder": "Enter message text here...",
                "size": undefined,
                "startAdornment": undefined,
                "sx": Object {
                  "flex": 1,
                  "ml": 1,
                },
                "type": "text",
              }
            }
            sx={
              Object {
                "flex": 1,
                "ml": 1,
              }
            }
          >
            <Noop />
            <div
              className="MuiInputBase-root MuiInputBase-colorPrimary css-piqts5-MuiInputBase-root"
              onClick={[Function]}
            >
              <MuiInputBaseInput
                aria-label="Enter message text here"
                as="input"
                autoComplete="off"
                className="MuiInputBase-input"
                data-testid="message"
                id="message"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                ownerState={
                  Object {
                    "autoComplete": "off",
                    "color": "primary",
                    "disabled": undefined,
                    "endAdornment": undefined,
                    "error": undefined,
                    "focused": false,
                    "formControl": undefined,
                    "fullWidth": false,
                    "hiddenLabel": undefined,
                    "id": "message",
                    "inputProps": Object {
                      "aria-label": "Enter message text here",
                      "data-testid": "message",
                    },
                    "multiline": false,
                    "placeholder": "Enter message text here...",
                    "size": undefined,
                    "startAdornment": undefined,
                    "sx": Object {
                      "flex": 1,
                      "ml": 1,
                    },
                    "type": "text",
                  }
                }
                placeholder="Enter message text here..."
                type="text"
              >
                <Noop />
                <input
                  aria-label="Enter message text here"
                  autoComplete="off"
                  className="MuiInputBase-input css-yz9k0d-MuiInputBase-input"
                  data-testid="message"
                  id="message"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  placeholder="Enter message text here..."
                  type="text"
                />
              </MuiInputBaseInput>
            </div>
          </MuiInputBaseRoot>
        </ForwardRef(InputBase)>
        <ForwardRef(IconButton)
          aria-label="Send"
          data-testid="send-message"
          sx={
            Object {
              "p": "10px",
            }
          }
          type="submit"
        >
          <MuiIconButtonRoot
            aria-label="Send"
            centerRipple={true}
            className="MuiIconButton-root MuiIconButton-sizeMedium"
            data-testid="send-message"
            disabled={false}
            focusRipple={true}
            ownerState={
              Object {
                "aria-label": "Send",
                "children": <Memo />,
                "color": "default",
                "data-testid": "send-message",
                "disableFocusRipple": false,
                "disabled": false,
                "edge": false,
                "size": "medium",
                "sx": Object {
                  "p": "10px",
                },
                "type": "submit",
              }
            }
            sx={
              Object {
                "p": "10px",
              }
            }
            type="submit"
          >
            <Noop />
            <ForwardRef(ButtonBase)
              aria-label="Send"
              centerRipple={true}
              className="MuiIconButton-root MuiIconButton-sizeMedium css-s8yeam-MuiIconButton-root"
              data-testid="send-message"
              disabled={false}
              focusRipple={true}
              type="submit"
            >
              <MuiButtonBaseRoot
                aria-label="Send"
                as="button"
                className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-s8yeam-MuiIconButton-root"
                data-testid="send-message"
                disabled={false}
                onBlur={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                ownerState={
                  Object {
                    "aria-label": "Send",
                    "centerRipple": true,
                    "children": <Memo />,
                    "className": "MuiIconButton-root MuiIconButton-sizeMedium css-s8yeam-MuiIconButton-root",
                    "component": "button",
                    "data-testid": "send-message",
                    "disableRipple": false,
                    "disableTouchRipple": false,
                    "disabled": false,
                    "focusRipple": true,
                    "focusVisible": false,
                    "tabIndex": 0,
                    "type": "submit",
                  }
                }
                tabIndex={0}
                type="submit"
              >
                <Noop />
                <button
                  aria-label="Send"
                  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1gtpq4r-MuiButtonBase-root-MuiIconButton-root"
                  data-testid="send-message"
                  disabled={false}
                  onBlur={[Function]}
                  onContextMenu={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="submit"
                >
                  <ForwardRef(SendIcon)>
                    <ForwardRef(SvgIcon)
                      data-testid="SendIcon"
                    >
                      <MuiSvgIconRoot
                        aria-hidden={true}
                        as="svg"
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium"
                        data-testid="SendIcon"
                        focusable="false"
                        ownerState={
                          Object {
                            "children": <path
                              d="M2.01 21 23 12 2.01 3 2 10l15 2-15 2z"
                            />,
                            "color": "inherit",
                            "component": "svg",
                            "data-testid": "SendIcon",
                            "fontSize": "medium",
                            "inheritViewBox": false,
                            "viewBox": "0 0 24 24",
                          }
                        }
                        viewBox="0 0 24 24"
                      >
                        <Noop />
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="SendIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M2.01 21 23 12 2.01 3 2 10l15 2-15 2z"
                          />
                        </svg>
                      </MuiSvgIconRoot>
                    </ForwardRef(SvgIcon)>
                  </ForwardRef(SendIcon)>
                  <ForwardRef(TouchRipple)
                    center={true}
                  >
                    <MuiTouchRippleRoot
                      className="MuiTouchRipple-root"
                    >
                      <Noop />
                      <span
                        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      >
                        <TransitionGroup
                          childFactory={[Function]}
                          component={null}
                          exit={true}
                        />
                      </span>
                    </MuiTouchRippleRoot>
                  </ForwardRef(TouchRipple)>
                </button>
              </MuiButtonBaseRoot>
            </ForwardRef(ButtonBase)>
          </MuiIconButtonRoot>
        </ForwardRef(IconButton)>
      </form>
    </MuiPaperRoot>
  </ForwardRef(Paper)>
</NewMessage>
`;
